src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1736,15 +1736,20 @@ protected override void InvokeDefaultAction(string path)
                                 return;
                             }
 
-
-                            bool hidden = false;
-                            if (!Force) hidden = (recursiveDirectory.Attributes & FileAttributes.Hidden) != 0;
-
-                            // if "Hidden" is explicitly specified anywhere in the attribute filter, then override
-                            // default hidden attribute filter.
-                            if (Force || !hidden || isFilterHiddenSpecified || isSwitchFilterHiddenSpecified)
+                            if (!InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(recursiveDirectory))
```

Comment:
@jeffbi Could you please summarize the description and add it as comments right before this statement? I hope the comment can explain why we are checking `IsReparsePoint` on `recursiveDirectory` and what behavior of `Get-ChildItem` we are trying to get by having this check. I'm sure the comments will be very helpful to other people when looking at this code.